library/src/tests/resources/src/state_preparation.qs

```diff
@@ -106,4 +106,23 @@ namespace Test {
         }
     }
 
+    operation TestPrepareUniformSuperposition(nStates : Int) : Unit {
+        use qs = Qubit[10];
+        PrepareUniformSuperposition(nStates, qs);
+        DumpMachine();
+        ResetAll(qs);
+    }
+
+    operation TestPrepareUniformSuperpositionExhaustive() : Unit {
+        let NBits = 4;
+        use qs = Qubit[NBits];
```

Comment:
Since the intent of each iteration of the loop is start from qubits in the zero state and assert they return to the zero state, you could move the allocation into the loop and let the qubits go out of scope between each iteration. That will also have the minor benefit of preventing any accumulation of global phase across iterations which can help with later investigations.

Reply:
Makes sense. Some part of me tries to save on allocation/release of qubits, but that may be wrong intuition. More allocations and release of qubits may pave a way to qubit reuse optimizations.